code/modules/modular_computers/file_system/programs/ntmessenger.dm

```diff
@@ -406,21 +407,32 @@
 /// topic call that answers to people pressing "(Reply)" in chat
 /datum/computer_file/program/messenger/Topic(href, href_list)
 	..()
+
 	if(QDELETED(src))
 		return
-	// send an activation message, open the messenger, kill whoever reads this nesting mess
-	if(!computer.enabled)
-		if(!computer.turn_on(usr, open_ui = FALSE))
-			return
-	if(computer.active_program != src)
-		if(!computer.open_program(usr, src, open_ui = FALSE))
-			return
-	if(!href_list["close"] && usr.can_perform_action(computer, FORBID_TELEKINESIS_REACH))
-		switch(href_list["choice"])
-			if("Message")
-				send_message(usr, list(locate(href_list["target"])))
-			if("mess_us_up")
-				if(!HAS_TRAIT(src, TRAIT_PDA_CAN_EXPLODE))
-					var/obj/item/modular_computer/pda/comp = computer
-					comp.explode(usr, from_message_menu = TRUE)
-					return
+
+	if(!computer.enabled && !computer.turn_on(usr, open_ui = FALSE))
+		return
+
+	if(computer.active_program != src && !computer.open_program(usr, src, open_ui = FALSE))
+		return
+
+	if(!issilicon(usr) && !usr.can_perform_action(computer, FORBID_TELEKINESIS_REACH))
+		return
+
+	var/reply_target = href_list["reply"]
+	if(!isnull(reply_target))
+		var/messenger = locate(reply_target)
```

Comment:
unscoped locate

Reply:
messenger = locate(reply_target) in GetViewableDevices(sort_by_job)
you might get away with removing the istype checks because only messengers are stored in the list. Not sure how TG standards are though.